### DISCUSSION TITLE
Device Authorization Flow

### DISCUSSION THREAD

**Discussion by imikemiller** [nextauthjs/next-auth#9779] [Help]
### Question 💬

I am struggling to find any useful information regarding a [device authorization flow.](https://datatracker.ietf.org/doc/html/rfc8628) I have a conventional nextjs application that uses nextauth and an out-of-the-box oAuth provider with the prisma adapter. I would like to provide a JWT token to a non-browser client (a CLI app) such that it it is able to talk to the nextauth protected API routes on the nextjs app. 

To do this I would need to allow some API routes to be authorized by using the request authorization header instead of sending the JWT as a cookie. I also need to be able to mint JWT tokens for logged in users and securely communicate them to the CLI app. I intended to do this using the Device Flow and configure my next app as an oauth2 server such that I can control this flow but lean on the existing authentication models. 

I dont want to roll my own auth alongside nextauth as it seems like this should be possible. Does anyone have any suggestions / direction for this problem? 

### How to reproduce ☕️

Its not reproducible 

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by NicHaley** [comment]
I have the same question

**Comment by smultar** [comment]
I use a database strategy, and then I create an endpoint that uses the app protocol `example://internalAddress/listenApi`.

From there an API key is generated and saved under my user. When the app makes API calls to my server, it checks for a session or a valid API key.

Then I store the API key internally on the app. It has an internal expiration date as well. My API will deny access on endpoints if the key is no longer valid.

**Comment by shahzeb1** [comment]
Is there a more formal / better approach to this. I wish the flow to document CLI -> next auth was documented.

@smultar do you happen to have a tiny MVP of a CLI that uses the flow you're referring to?
